neo/Network/LocalNode.cs

```diff
@@ -437,9 +462,14 @@ public void Dispose()
                         nodes = connectedPeers.ToArray();
                     }
                     Task.WaitAll(nodes.Select(p => Task.Run(() => p.Disconnect(false))).ToArray());
+
                     new_tx_event.Set();
                     if (poolThread?.ThreadState.HasFlag(ThreadState.Unstarted) == false)
                         poolThread.Join();
+                    
+                    // Need to ensure any outstanding calls to Blockchain_PersistCompleted are not in progress.
+                    // TODO: could add locking instead of using an arbitrarily long sleep here.
+                    Thread.Sleep(3000);
```

Comment:
Do you have any plans to improve this place before this PR is merged?

Reply:
Yes, I should be able to adjust this to obtain the new PersistLock now. I’ll adjust and test it and update the pull request.

Reply:
Should be good now.